app/services/initializable.rb

```diff
@@ -0,0 +1,8 @@
+module Initializable
+  def initialize(params = {})
+    params.each do |key, value|
+      setter = "#{key}="
```

Comment:
Would you elaborate on this function? I don't understand what's going on here. Are some of the params just methods of the particular object?

Reply:
So this is a ruby meta-programming method, basically anytime that you call initialize with a series of params like so
`Object.new(id: "1234", some_property: "foobar")`

Then for each of those params we'll iterate those and call the associated setter method. In this example this method would call the methods `id=`, and `some_property=`.

Now, this is where it kind of gets strange because you'll notice in `user.rb` and `card.rb` I do not define any methods like this. 

That's where `attr_accessor` comes in. This method allows for a shorthand for generating getters/setters. So if I have a class like 
```ruby
class Card
  def id
    @name
  end

  def id=(str)
    @id = str
  end
end
```

This can be shortened to
```ruby
class Card
  attr_accessor :id
end
```

So in this method we go through each of the passed hash params, define a string called `setter` for the method we want to call with our value being the hash value.

So in our example above 

```ruby
# passed in through the params
value = "1234" 
# the property we're setting
setter = "id=" 
# send the method the value
send(setter, value)
# calls the method id=("1234")
```

Reply:
Cool that makes sense. Would be good to add a small comment with your example.

Reply:
https://github.com/brave-intl/publishers/pull/2062/commits/0126df6ebaec13b1e1bfc8c6f9ae98517a045e65